test/close.test.js

```diff
@@ -291,3 +291,38 @@ test('Cannot be reopened the closed server has listen callback', async t => {
     t.ok(err)
   })
 })
+
+test('shutsdown while keep-alive connections are active (non-async)', t => {
+  t.plan(5)
+
+  const timeoutTime = 2 * 60 * 1000
+  const fastify = Fastify({ forceCloseConnections: true })
+
+  fastify.server.setTimeout(timeoutTime)
+  fastify.server.keepAliveTimeout = timeoutTime
+
+  fastify.get('/', (req, reply) => {
+    reply.send({ hello: 'world' })
+  })
+
+  fastify.listen(0, (err, address) => {
+    t.error(err)
+
+    const client = new Client(address, { keepAliveTimeout: 1 * 60 * 1000 })
+    client.request({ path: '/', method: 'GET' }, (err, response) => {
+      t.error(err)
+      t.equal(client.closed, false)
+
+      fastify.close((err) => {
+        t.error(err)
+
+        // Due to the nature of the way we reap these keep-alive connections,
+        // there hasn't been enough time before the server fully closed in order
+        // for the client to have seen the socket get destroyed. The mere fact
+        // that we have reached this callback is enough indication that the
+        // feature being tested works as designed.
+        t.equal(client.closed, false)
```

Comment:
Should the client receive an `error` event due to the socket being closed?

Reply:
Maybe?